notebook/agentchat_postresponse_secret_redaction.ipynb

```diff
@@ -0,0 +1,309 @@
+{
+	"cells": [{
+			"cell_type": "markdown",
+			"metadata": {},
+			"source": [
+				"# Redact Sensitive Information Post-Response with Hooks\n",
+				"\n",
+				"This notebook shows an example on how we can use hooks in agents to effectively hide/redact sensitive data(i.e. API keys) from getting printed out to avoid potential data leak from agent's responses\n",
+				"\n",
+				"## Requirements\n",
+				"\n",
+				"AutoGen requires `Python>=3.8`. To run this notebook example, please install pyautogen:\n",
+				"```bash\n",
+				"pip install pyautogen\n",
+				"```"
+			]
+		},
+		{
+			"cell_type": "code",
+			"execution_count": null,
+			"metadata": {},
+			"outputs": [],
+			"source": [
+				"from typing import Dict, Union\n",
+				"import os\n",
+				"import copy\n",
+				"\n",
+				"import autogen\n",
+				"from autogen import ConversableAgent\n",
+				"from autogen.coding import LocalCommandLineCodeExecutor\n",
+				"from autogen.code_utils import create_virtual_env"
+			]
+		},
+		{
+			"cell_type": "markdown",
+			"metadata": {},
+			"source": [
+				"# Set Environment Data"
+			]
+		},
+		{
+			"cell_type": "code",
+			"execution_count": null,
+			"metadata": {},
+			"outputs": [],
+			"source": [
+				"os.environ['VENV_DIR'] = \"<Your venv directory that is used for code executor agent>\"\n",
+				"os.environ['TEMP_FILE_DIR'] = \"<Your directory for temporary code files>\"\n",
+				"os.environ['LLM_API_KEY'] = \"<Your API key for llm config>\"\n",
+				"os.environ['LLM_API_TYPE'] = \"<Your API type for llm config>\"\n",
+				"os.environ['LLM_API_VERSION'] = \"<Your API version for llm config>\"\n",
+				"os.environ['LLM_API_URL'] = \"<Your API Url for llm config>\""
+			]
+		},
+		{
+			"cell_type": "code",
+			"execution_count": 7,
+			"metadata": {},
+			"outputs": [],
+			"source": [
+				"# Get the env variables\n",
+				"venv_dir = os.environ['VENV_DIR']\n",
+				"temp_file_fir = os.environ['TEMP_FILE_DIR'] \n",
+				"\n",
+				"llm_config = {\n",
+				"    \"config_list\": [{\n",
+				"        \"model\": \"gpt-4\", \n",
+				"        \"api_key\": os.environ[\"LLM_API_KEY\"], \n",
+				"        \"api_type\": os.environ[\"LLM_API_TYPE\"], \n",
+				"        \"api_version\": os.environ[\"LLM_API_VERSION\"], \n",
+				"        \"base_url\": os.environ[\"LLM_API_URL\"]}],\n",
+				"}\n"
+			]
+		},
+		{
+			"cell_type": "code",
+			"execution_count": 8,
+			"metadata": {},
+			"outputs": [],
+			"source": [
+				"# The string that sensitive data will be redated to\n",
+				"replacementString = \"REDACTED\""
+			]
+		},
+		{
+			"cell_type": "code",
+			"execution_count": 9,
+			"metadata": {},
+			"outputs": [],
+			"source": [
+				"def get_all_env_secrets():\n",
+				"    '''\n",
+				"        Gets all env variables and return them as a dictionary\n",
+				"    '''\n",
+				"    all_env_secrets = os.environ\n",
+				"\n",
+				"    return all_env_secrets"
```

Comment:
Why is this function required? you can simply call `os.environ` wherever it's needed